_studio/mfx_lib/lpla/mfx_lp_lookahead.cpp

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2014-2020 Intel Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "mfx_lp_lookahead.h"
+#include "mfx_h265_encode_hw.h"
+#include "mfx_h265_encode_hw_utils.h"
+#include "mfx_vpp_main.h"
+#include "libmfx_core_interface.h"
+
+#if defined (MFX_ENABLE_LP_LOOKAHEAD)
+
+#define MSDK_ALIGN16(value)             (((value + 15) >> 4) << 4)
+#define MSDK_MEMCPY_VAR(dstVarName, src, count) memcpy(&(dstVarName), (src), (count))
```

Comment:
I don't think you need it. operator= for structs doing the same avoiding low-level memory operations